CHANGELOG.md

```diff
@@ -1,5 +1,135 @@
 # CHANGELOG
 
+# Version 3.4.0-BETA
+
+Migration notes:
+
+* Rate-limited sampling is the new default. If you have not configured a sampling percentage
+  and are using the prior default behavior of 100%, you will get the new default which limits
+  the total requests captured to (approximately) 5 requests per second. If you wish to continue with
+  the previous behavior, you can explicitly specify a sampling percentage of 100, e.g.
+  ```
+  {
+    "sampling": {
+      "percentage": 100
+    }
+  }
+  ```
+* Both percentage-based and rate-limited sampling now only apply to requests, and to telemetry that
+  is captured in the context of a request.
+  If you do want to sample "standalone" telemetry (e.g. startup logs), you can use sampling
+  overrides, e.g.
+  ```
+  {
+    "preview": {
+      "sampling": {
+        "overrides": [
+          {
+            "telemetryType": "dependency",
+            "includeStandaloneTelemetry": true,
+            "percentage": 10
+          }
+        ]
+      }
+    }
+  }
+  ```
+
+Enhancements:
+
+* Standard metrics for HTTP requests and HTTP dependencies are now pre-aggregated inside of the Java
+  agent, and so they are no longer affected by sampling
+  ([#2439](https://github.com/microsoft/ApplicationInsights-Java/pull/2439)).
+* Rate-limited sampling has been introduced which can be used to tune ingestion costs
+  ([#2456](https://github.com/microsoft/ApplicationInsights-Java/pull/2456)), e.g.
+  ```
+  {
+    "sampling": {
+      "limitPerSecond": 5
+    }
+  }
+  ```
+  Note: the `limitPerSecond` can be a decimal value, including values less than 1.
+* Exceptions are no longer captured directly on dependency records, in order to reduce ingestion
+  cost and because exceptions which are logged or bubble up to the request are already captured
+  ([#2423](https://github.com/microsoft/ApplicationInsights-Java/pull/2423)).
+* New versions of the `applicationinsights-core` and `applicationinsights-web` 2.x SDK artifacts
+  have been released, with a reduced API surface area to makes it clear which APIs interop with the
+  3.x Java agent ([#2418](https://github.com/microsoft/ApplicationInsights-Java/pull/2418)).
+  These new versions are no-ops when run without the Java agent. To update your dependencies:
+  | 2.x dependency | Action | Remarks |
+  |----------------|--------|---------|
+  | `applicationinsights-core` | Update the version to `3.4.0` or later | |
```

Comment:
this is beta 3.4.0-BETA. should we mention the new shim till 3.4.0 GA?

Reply:
I think it's good to include here, hopefully someone will give it a spin during BETA, I will update the version mentioned here though